feat: Реализован компонент профиля пользователя и сервис для работы с…

mutaliev2008 opened the pull request:
… пользователями.

mutaliev2008 committed:
feat: Реализован компонент профиля пользователя и сервис для работы с пользователями.

Mlovz commented on src/app/core/models/user.model.ts:
```diff
@@ -0,0 +1,6 @@
+export interface TUser {
```
зачем в названии интерфейса указывать T? можно просто User

mutaliev2008 replied:
В Яндексе просили так писать типо профессионально

Mlovz commented on src/app/core/services/user.service.ts:
```diff
@@ -0,0 +1,30 @@
+  providedIn: 'root',
+})
+export class UserService {
+  users: TUser[] = usersData;
```
тут массив users изначально должен быть пустым, а в getAllUsers присваивать userData в users

Mlovz commented on src/app/core/services/user.service.ts:
```diff
@@ -0,0 +1,30 @@
+    console.log(`Пользователь с id ${id} обновлен.`, this.users);
+  }
+  switchUser() {
+    this.users = this.users.reverse();
```
reverse ?

Mlovz commented on src/app/features/pages/profile-component/profile.component.ts:
```diff
@@ -0,0 +1,19 @@
+})
+export class ProfileComponent implements OnInit {
+  private userService = inject(UserService);
+  users: TUser[] = [];
```
возьми юзеров из сервиса, тут не нужно их еще раз создавать, они у тя уже лежат в сераисе

mutaliev2008 committed:
feat: Реализован выбор и отображение данных активного пользователя + стилизация кнопки.

Изменения внесены в:
- ProfileComponent (логика выбора и отображения)
- UserService (получение данных пользователя)
- ActiveButtonColorDirective (стилизация активной кнопки)

mutaliev2008 committed:
feat(pipe, directive, html)
